nomenclature/definition.py

```diff
@@ -136,22 +146,67 @@ def check_aggregate(self, df: IamDataFrame, **kwargs) -> None:
             error = pd.concat(lst)
             return error if not error.empty else None
 
-    def to_excel(
-        self, excel_writer, sheet_name=None, sort_by_code: bool = False, **kwargs
-    ):
-        """Write the *variable* codelist to an Excel sheet
+    def to_excel(self, excel_writer, sort_by_code: bool = False, **kwargs):

```

Comment:
I would simplify the `excel_writer` input to just be a str or path object. I don't think the use case where we would want to feed in an ExcelWriter object really exists.
To further simplify I would also take out the `sort_by_code` input and manually set it to true for all code lists. 
Also if there's no concrete use case for the kwargs, I would take them out as well.

Reply:
Agree on the simplificatons, but let's leave in the kwargs - they are also less ambiguous now thanks to the simplification.